tsconfig.json

```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
-    "module": "CommonJS",
-    "target": "ES2019",
+    "module": "ES2020",
+    "target": "ESNext",
```

Comment:
Targeting ESNext might emit code that is incompatible with every current runtime[^1] since TS will allow syntax under proposal or accepted for future releases.
I think we should either keep ES2020 or at most bump to ES2022, browser baseline 2025 still doesn't include every feature from ES2024 and keeping one year gap sounds like a good idea for a client lib.

[^1]: https://node.green/#ESNEXT and https://caniuse.com/mdn-javascript_builtins_regexp_escape

Reply:
Agreed, we can stick to `ES2020`.